From 705c794a5f1bf4100e0a5d05c4e39f90fe856b7d Mon Sep 17 00:00:00 2001
From: "Angelo G. Del Regno" <kholk11@gmail.com>
Date: Fri, 27 Oct 2017 11:09:58 +0200
Subject: [PATCH] msm8998: vdec: Add missing ifdefs for UBWC on DPB buffer
 decision

When the UBWC feature is available on a SoC, we define _UBWC_.
Vice-versa, when this is not defined, do not try to set any
UBWC mode as, otherwise, the Venus firmware may crash because
of unsupported modes, breaking the whole vidc functionality.

This happens on at least "legacy" SoCs, such as MSM8917,
MSM8939, MSM8952, MSM8956, MSM8976.

Change-Id: I76131db5272b97016679c5bc0bf6ae099167cd03
---
 msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp b/msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
index 7081cf6..3b22632 100644
--- a/msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
+++ b/msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
@@ -1065,6 +1065,7 @@ OMX_ERRORTYPE omx_vdec::decide_dpb_buffer_mode(bool split_opb_dpb_with_same_colo
         dpb_bit_depth == MSM_VIDC_BIT_DEPTH_10;
     bool dither_enable = true;
 
+#ifdef _UBWC_
     switch (m_dither_config) {
     case DITHER_DISABLE:
         dither_enable = false;
@@ -1098,7 +1099,7 @@ OMX_ERRORTYPE omx_vdec::decide_dpb_buffer_mode(bool split_opb_dpb_with_same_colo
         }
 
     }
-
+#endif /* _UBWC_ */
 
     if (!BITMASK_PRESENT(&m_flags ,OMX_COMPONENT_IDLE_PENDING) &&
         !BITMASK_PRESENT(&m_flags, OMX_COMPONENT_OUTPUT_ENABLE_PENDING)) {
@@ -1106,7 +1107,7 @@ OMX_ERRORTYPE omx_vdec::decide_dpb_buffer_mode(bool split_opb_dpb_with_same_colo
         return eRet;
     }
 
-
+#ifdef _UBWC_
     if (cpu_access) {
         if (dpb_bit_depth == MSM_VIDC_BIT_DEPTH_8) {
             /* Disabled split mode for VP9. In split mode the DPB buffers are part of the internal
@@ -1170,6 +1171,11 @@ OMX_ERRORTYPE omx_vdec::decide_dpb_buffer_mode(bool split_opb_dpb_with_same_colo
             }
         }
     }
+#else
+    eRet = set_dpb(split_opb_dpb_with_same_color_fmt,
+                       V4L2_MPEG_VIDC_VIDEO_DPB_COLOR_FMT_NONE);
+#endif /* _UBWC_ */
+
     if (eRet) {
         DEBUG_PRINT_HIGH("Failed to set DPB buffer mode: %d", eRet);
     }
-- 
2.15.1

